examples/src/main/java/io/grpc/examples/preserialized/PreSerializedServer.java

```diff
@@ -0,0 +1,159 @@
+      HelloReply reply = HelloReply.newBuilder().setMessage("Hello " + req.getName()).build();
+      responseObserver.onNext(reply.toByteArray());
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public ServerServiceDefinition bindService() {
+      MethodDescriptor<HelloRequest, HelloReply> sayHello = GreeterGrpc.getSayHelloMethod();
+      // Modifying the method descriptor to use bytes as the response, instead of HelloReply. By
+      // adjusting toBuilder() you can choose which of the request and response are bytes.
+      MethodDescriptor<HelloRequest, byte[]> byteSayHello = sayHello
+          .toBuilder(sayHello.getRequestMarshaller(), new ByteArrayMarshaller())
+          .build();
+      // GreeterGrpc.bindService() will bind every service method, including sayHello(). (Although
+      // Greeter only has one method, this approach would work for any service.) AsyncService
+      // provides a default implementation of sayHello() that returns UNIMPLEMENTED, and that
+      // implementation will be used by bindService(). replaceMethod() will rewrite that method to
+      // use our byte-based method instead.
+      //
+      // The generated bindService() uses ServerCalls to make RPC handlers. Since the generated
+      // bindService() won't expect byte[] in the AsyncService, this uses ServerCalls directly. It
+      // isn't as convenient, but it behaves the same as a normal RPC handler.
+      return replaceMethod(
+          GreeterGrpc.bindService(this),
+          byteSayHello,
+          ServerCalls.asyncUnaryCall(this::byteSayHello));
+    }
+
+    /** Rewrites the ServerServiceDefinition replacing one method's definition. */
+    private static <ReqT, RespT> ServerServiceDefinition replaceMethod(
+        ServerServiceDefinition def,
+        MethodDescriptor<ReqT, RespT> desc,
+        ServerCallHandler<ReqT, RespT> handler) {
+      // There are two data structures involved. The first is the "descriptor" which describes the
+      // service and methods as a schema. This is the same on client and server. The second is the
+      // "definition" which includes the handlers to execute methods. This is specific to the server
+      // and is generated by "bind." This adjusts both the descriptor and definition.
+      ServiceDescriptor srvDesc = def.getServiceDescriptor();
+      ServiceDescriptor.Builder srvBuilder = ServiceDescriptor.newBuilder(srvDesc.getName())
+          .setSchemaDescriptor(srvDesc.getSchemaDescriptor())
+          .addMethod(desc); // Add the modified method
+      // Copy methods other than the modified one
+      for (MethodDescriptor<?,?> md : srvDesc.getMethods()) {
+        if (desc.getFullMethodName().equals(md.getFullMethodName())) {
+          continue;
+        }
+        srvBuilder.addMethod(md);
+      }
+      ServerServiceDefinition.Builder b = ServerServiceDefinition.builder(srvBuilder.build())
```

Comment:
Insert a comment indicating that you are moving from constructing the ServiceDescriptor to constructing the ServerServiceDefinition.